### PR TITLE
Fix #7763 gpx track selection

### LINKED ISSUE #7763: Track screen's folder tick marks do not refllect selected gpx file status
How to reproduce:
1. Orgamize some of your gpx files in folders (under OsmAnd's 'tracks' folder)
2. On the My places / tracks screen, then tap the 'Show on map' button near the bottom center of the screen, this will produce selection tick marks in front of all gpx files and folders.
3. Select the tick mark in front of one of the subfoldets. This will correctly select and display all tracks contained.
4. Now leave the screen and later return to it: The tickmark of the selected folde remains empty, although all contained files are corrently selected.

 The folder tick mark must correctly dhow the display status of selected gpx tracks, preferably even indicate by perhaps gray vs. lack checkmark color if all or just some contained files are selected to be displayed.visible.

### PR DESCRIPTION
**Pull request by Dima-1**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/myplaces/AvailableGPXFragment.java [hunk 1]
```diff
@@ -605,19 +625,22 @@ public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
 
 			@Override
 			public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
-				runSelection(false);
+				HashMap<String, Boolean> selectedItemsFileNames = new HashMap<>();
+				originalSelectedItems.addAll(selectedItems);
+				for (GpxInfo gpxInfo : originalSelectedItems) {
+					if (gpxInfo.currentlyRecordingTrack) {
+						selectedItemsFileNames.put(CURRENT_TRACK, true);
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/myplaces/AvailableGPXFragment.java]: Why always true for the current track?

File: OsmAnd/src/net/osmand/plus/GpxSelectionHelper.java [hunk 2]
```diff
@@ -54,7 +56,12 @@
 	private Map<GPXFile, Long> selectedGpxFilesBackUp = new java.util.HashMap<>();
 	private SavingTrackHelper savingTrackHelper;
 	private final static Log LOG = PlatformUtil.getLog(GpxSelectionHelper.class);
+	private SelectGpxTask selectGpxTask;
+	private SelectGpxTaskListener gpxTaskListener;
 
+	public void setGpxTaskListener(SelectGpxTaskListener gpxTaskListener) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/GpxSelectionHelper.java]: No field is needed ! Fields should be created carefully and when it is not needed better to use parameter for method

File: OsmAnd/src/net/osmand/plus/GpxSelectionHelper.java [hunk 3]
```diff
@@ -698,6 +715,10 @@ public void setGpxFile(GPXFile gpxFile, OsmandApplication app) {
 			processPoints(app);
 		}
 
+		public boolean isLoaded() {
+			return gpxFile.modifiedTime != 0;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/GpxSelectionHelper.java]: better to use -1 if it is going to be used. Sometimes 0 is system time

File: OsmAnd/src/net/osmand/plus/GpxSelectionHelper.java [hunk 4]
```diff
@@ -54,7 +56,12 @@
 	private Map<GPXFile, Long> selectedGpxFilesBackUp = new java.util.HashMap<>();
 	private SavingTrackHelper savingTrackHelper;
 	private final static Log LOG = PlatformUtil.getLog(GpxSelectionHelper.class);
+	private SelectGpxTask selectGpxTask;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/GpxSelectionHelper.java]: Field is not used anywhere

Comment (Chumva) [reply OsmAnd/src/net/osmand/plus/GpxSelectionHelper.java]: This field is used for checking running SelectGpxTask in runSelection

File: OsmAnd/src/net/osmand/plus/GpxSelectionHelper.java (hunk 2 above)
Comment (Chumva) [reply OsmAnd/src/net/osmand/plus/GpxSelectionHelper.java]: I know, but with this listener, we can get updates for necessary screens if update working in the background and it doesn`t update new screen.

**Comment by vshcherb** [comment]
Fix #7763

**Pull request merged**